tracehandler/tracehandler.go

```diff
@@ -79,27 +77,20 @@ type traceHandler struct {
 // newTraceHandler creates a new traceHandler
 func newTraceHandler(rep reporter.TraceReporter, traceProcessor TraceProcessor,
 	intervals Times, cacheSize uint32) (*traceHandler, error) {
-	bpfTraceCache, err := lru.New[host.TraceHash, libpf.TraceHash](
+	traceCache, err := lru.New[host.TraceHash, *libpf.Trace](
 		cacheSize, func(k host.TraceHash) uint32 { return uint32(k) })
 	if err != nil {
 		return nil, err
 	}
+	// Do not hold elements indefinitely in the cache.
+	traceCache.SetLifetime(traceCacheLifetime)
```

Comment:
Is the purpose of the lifetime to get rid of cache entries that haven't been used since `traceCacheLifetime`? Then you should regularly call `PurgeExpired()` and possibly use NewSynced instead of New in case that introduces concurrency.

Reply:
Switched to SyncedLRU with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/405/commits/da54892ba182e460c1177f011f27fcb2d78c69cc.